scopesim/source/source.py

```diff
@@ -601,20 +618,10 @@ def __add__(self, new_source):
     def __radd__(self, new_source):
         return self.__add__(new_source)
 
-    def __repr__(self):
-        msg = ""
-        for ifld, fld in enumerate(self.fields):
-            if isinstance(fld, Table):
-                tbl_len = len(fld)
-                num_spec = set(fld["ref"])
-                msg += f"[{ifld}]: Table with {tbl_len} rows, referencing spectra {num_spec} \n"
-            elif isinstance(fld, (fits.ImageHDU, fits.PrimaryHDU)):
-                im_size = fld.data.shape if fld.data is not None else "<empty>"
-                num_spec = "-"
-                msg += f"[{ifld}]: ImageHDU with size {im_size}"
-                if "SPEC_REF" in self.fields[ifld].header:
-                    num_spec = self.fields[ifld].header["SPEC_REF"]
-                    msg += f", referencing spectrum {num_spec}"
-                msg += "\n"
-
-        return msg
+    def __repr__(self) -> str:
+        with StringIO() as str_stream:
+            for ifld, fld in enumerate(self.fields):
+                str_stream.write(f"[{ifld}]: ")
+                fld._write_stream(str_stream)
+                str_stream.write("\n")
+            return str_stream.getvalue()
```

Comment:
This seems overly complicated and not Pythonic. Perhaps:
```suggestion
    def __repr__(self) -> str:
        s_fields = [
            f"[{ifld}]: {repr(fld)}\n"
            for ifld, fld in enumerate(self.fields)
        ]
        return "".join(s_fields)
```
And then implement `__repr__()` in the source field classes instead of `_write_stream()`.